watershed_workflow/river_tree.py

```diff
@@ -206,6 +223,420 @@ def construct_rivers_by_hydroseq(cls, segments):
         return roots
     
 
+    def deep_copy(self):
+        cp = copy.deepcopy(self)
+        for node1,node2 in zip(cp.preOrder(),self.preOrder()):
+            node1.properties = copy.deepcopy(node2.properties)
+        return cp
+
+
+def sort_children_by_angle(tree, reverse=False):
+    """Sorts the children of a given segment by their angle with respect to that segment."""
+    for node in tree.preOrder():
+        if len(node.children) > 1:
+            # compute tangents
+            my_seg_tan = np.array(node.segment.coords[0]) - np.array(node.segment.coords[1])
+
+            if reverse: sign = -1
+            else: sign = 1
 
+            def angle(c):
+                tan = np.array(c.segment.coords[-2]) - np.array(c.segment.coords[-1])
+                return sign*watershed_workflow.utils.angle(my_seg_tan, tan)
+           
+            node.children.sort(key=angle)
 
 
+def create_rivers_meshes(rivers, widths=8, enforce_convexity=True):
```

Comment:
Can these be moved out of this file and into a new file or into your new "triangulate" file?  I would like to keep "discretization" code (e.g. triangulate, etc) separate from "data representation" code (e.g. RiverTree).

Reply:
You mean only this parent function or other functions too like "create_river_corridor", "to_quads" etc?

Reply:
we talked about this, so hopefully it is clear now. 

Just split out things that make sense to be considered "mesh" things from things that are "river" things.

Reply:
the whole river meshing part of the code has been moved into a new file "create_river_mesh.py" . Tests and examples have been updated too